internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1133,6 +1162,40 @@ func (r *BMACReconciler) newSpokeMachine(bmh *bmh_v1alpha1.BareMetalHost, cluste
 		},
 	}
 	mutateFn := func() error {
+		if machine.ObjectMeta.Annotations == nil {
+			machine.ObjectMeta.Annotations = make(map[string]string)
+		}
+		machine.ObjectMeta.Annotations[BMH_ANNOTATION] = fmt.Sprintf("%s/%s", OPENSHIFT_MACHINE_API_NAMESPACE, bmh.Name)
+		providerSpecValueFormat := `{
```

Comment:
I'm not sure using the values from the master machine is what we want here. I understand these are not being "consumed" but I think the image used for worker nodes is different from the master's and I would rather have this value represent the real state of affairs. Considering that BMAC is the one creating the resource, I think we don't really have an excuse to add the right value here.